lib/bundler/cli/doctor.rb

```diff
@@ -86,9 +88,41 @@ def run
           end
         end.flatten.sort.each {|m| message += m }
         raise ProductionError, message
-      else
+      elsif !permissions_valid
         Bundler.ui.info "No issues found with the installed bundle"
       end
     end
+
+  private
+
+    def check_home_permissions
+      require "find"
+      files_not_readable_or_writable = []
+      files_not_owned_by_current_user_but_still_rw = []
+      Find.find(Bundler.home.to_s).each do |f|
+        if !File.writable?(f) || !File.readable?(f)
```

Comment:
My only concern here is that a file can be not readable/writable and also be owned by a different user.

Reply:
@segiddin, @colby-swandale I can add that functionality if it's needed

Reply:
Yes please